packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -182,10 +182,183 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
+   * DescribeStream, GetRecords, GetShardIterator, ListStreams.
+   * @param grantee The principal to grant access to
+   */
+  grantStreamRead(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM principal all data write operations to this table:
+   * BatchWriteItem, PutItem, UpdateItem, DeleteItem.
+   * @param grantee The principal to grant access to
+   */
+  grantWriteData(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM principal to all data read/write operations to this table.
+   * BatchGetItem, GetRecords, GetShardIterator, Query, GetItem, Scan,
+   * BatchWriteItem, PutItem, UpdateItem, DeleteItem
+   * @param grantee The principal to grant access to
+   */
+  grantReadWriteData(grantee: iam.IGrantable): iam.Grant;
+}
+
+/**
+ * Reference to a dynamodb table.
+ */
+export interface TableAttributes {
+  /**
+   * The ARN of the dynamodb table.
+   *
+   * @default no table arn
+   */
+  readonly tableArn?: string;
+
+  /**
+   * The table name of the dynamodb table.
+   *
+   * @default no table name
+   */
+  readonly tableName?: string;
+}
+
+abstract class TableBase extends Resource implements ITable {
+  /**
+   * @attribute
+   */
+  public abstract readonly tableArn: string;
+
+  /**
+   * @attribute
+   */
+  public abstract readonly tableName: string;
+
+  /**
+   * Adds an IAM policy statement associated with this table to an IAM
+   * principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:PutItem", "dynamodb:GetItem", ...)
+   */
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+        Lazy.stringValue({ produce: () => this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE })
+      ],
+      scope: this,
+    });
+  }
+
+  /**
+   * Adds an IAM policy statement associated with this table's stream to an
+   * IAM principal's policy.
+   * @param _grantee The principal (no-op if undefined)
+   * @param _actions The set of actions to allow (i.e. "dynamodb:DescribeStream", "dynamodb:GetRecords", ...)
+   */
+  public grantStream(_grantee: iam.IGrantable, ..._actions: string[]): iam.Grant {
+    throw new Error(`grantStream is not supported.`);
+  }
+
+  /**
+   * Permits an IAM principal all data read operations from this table:
+   * BatchGetItem, GetRecords, GetShardIterator, Query, GetItem, Scan.
+   * @param grantee The principal to grant access to
+   */
+  public grantReadData(grantee: iam.IGrantable) {
+    return this.grant(grantee, ...READ_DATA_ACTIONS);
+  }
+
+  /**
+   * Permits an IAM Principal to list streams attached to current dynamodb table.
+   *
+   * @param _grantee The principal (no-op if undefined)
+   */
+  public grantTableListStreams(_grantee: iam.IGrantable): iam.Grant {
+    throw new Error(`grantTableListStreams is not supported.`);
```

Comment:
Make this `abstract` instead of this error implementation, and make it throw in `Import`.